iso-codes.yaml

```diff
@@ -0,0 +1,8 @@
+name: iso-codes
+sources:
+  - type: archive
+    url: https://salsa.debian.org/iso-codes-team/iso-codes/-/archive/iso-codes-4.1/iso-codes-iso-codes-4.1.tar.gz
+    sha256: dd91ecc7cce92272e6892dc49f6f5c1db6fabc1ea7fe41ad98b7fa26c7b2c5dd
+  - type: shell
+    commands:
+      - autoreconf -vfi;
```

Comment:
I'm sure I will regret asking but what's with these semicolons at the end of the lines everywhere?

Reply:
@barthalion 

> I'm sure I will regret asking but what's with these semicolons at the end of the lines everywhere?

It would be nice if you could stop your passive-aggressive attitude.

The point is that I would like to group atomic operations into a single entry.

For example, installing icons should be treated as an atomic operation.

To do this in YAML, I have to use a semicolon to separate individual lines, because this is actually treated as a single line command:
```
  - icon_in="resources/icons/ocrfeeder.svg";
    icon_out="ocrfeeder.png";
    for s in {16,22,24,32,48,64,72,96,128,192,256,512}; do
    [[ ! -f "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/${icon_out}" ]] || continue;
    rsvg-convert "${icon_in}" -w "${s}" -h "${s}" -a -f png -o "${icon_out}";
    install -p -D -m 0644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/";
    done;
    install -p -D -m 0644 "${icon_in}" -t "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/";
```

An alternative would be to use the following syntax:
```
  - |
    icon_in="resources/icons/ocrfeeder.svg" &&
    icon_out="ocrfeeder.png" &&
    for s in {16,22,24,32,48,64,72,96,128,192,256,512}; do
      if [[ -f "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/${icon_out}" ]]; then
        continue
      fi &&
      rsvg-convert "${icon_in}" -w "${s}" -h "${s}" -a -f png -o "${icon_out}" &&
      install -p -D -m 0644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/"
    done &&
    install -p -D -m 0644 "${icon_in}" -t "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/"
```
However, I don't like it because we're wasting one line here (the first one), and it is less readable in my opinion (especially for loops).

An another alternative would be to use literally just single line command.
However, it only obscures the code and it definitely doesn't improve readability in any way.

Of course, there is no need to use a semicolon at the end of just single command. However, I would like to use the same syntax wherever possible.
A lot of things can be done in many different ways and everyone has their own coding style. I hope you will finally respect mine.

See also:
 * https://github.com/flathub/me.kozec.syncthingtk/blob/365c147a0a4eb1a7489ccf43adb7dc1042d2439a/me.kozec.syncthingtk.yaml#L55-L58
 * https://github.com/flathub/org.subsurface_divelog.Subsurface/blob/30ea6400994f0e983ff9086edcd8b08db3587e0d/org.subsurface_divelog.Subsurface.yaml#L128-L132
 * https://pastebin.com/jVTYqKxc

Reply:
In anticipation of the question, in this project I used my experimental tool for generating modules and manifest scaffolding. Perhaps I will publish the code when I finish work on dependency tracking.

Reply:
Correct, I definitely regret asking. Moving on, with Joaquim on board and you ignoring any requests anyway, I'll just let it go.